include/logging/log.h

```diff
@@ -0,0 +1,248 @@
+ *
+ * @param data   Pointer to the data to be logged.
+ * @param length Length of data (in bytes).
+ */
+#define LOG_HEXDUMP_WARN(data, length) \
+	LOG_INTERNAL_HEXDUMP_MODULE(LOG_LEVEL_WARN, data, length)
+
+/**
+ * @brief Writes an INFO level message to the log.
+ *
+ * @details It's meant to write generic user oriented messages.
+ *
+ * @param data   Pointer to the data to be logged.
+ * @param length Length of data (in bytes).
+ */
+#define LOG_HEXDUMP_INFO(data, length) \
+	LOG_INTERNAL_HEXDUMP_MODULE(LOG_LEVEL_INFO, data, length)
+
+/**
+ * @brief Writes an DEBUG level message to the log.
+ *
+ * @details It's meant to write developer oriented information.
+ *
+ * @param data   Pointer to the data to be logged.
+ * @param length Length of data (in bytes).
+ */
+#define LOG_HEXDUMP_DBG(data, length) \
+	LOG_INTERNAL_HEXDUMP_MODULE(LOG_LEVEL_DBG, data, length)
+
+/**
+ * @brief Writes an ERROR level hexdump message to the log.
+ *
+ * @details It's meant to report severe errors, such as those from which it's
+ * not possible to recover.
+ *
+ * @param inst   Pointer to the log filter associated with the instance.
+ * @param data   Pointer to the data to be logged.
+ * @param length Length of data (in bytes).
+ */
+#define LOG_INST_HEXDUMP_ERR(inst, data, length) \
+	LOG_INTERNAL_HEXDUMP_INSTANCE(LOG_LEVEL_ERR, inst, data, length)
+
+/**
+ * @brief Writes an WARNING level message to the log.
+ *
+ * @details It's meant to register messages related to unusual situations that
+ * are not necessarily errors.
+ *
+ * @param inst   Pointer to the log filter associated with the instance.
+ * @param data   Pointer to the data to be logged.
+ * @param length Length of data (in bytes).
+ */
+#define LOG_INST_HEXDUMP_WARN(inst, data, length) \
+	LOG_INTERNAL_HEXDUMP_INSTANCE(LOG_LEVEL_WARN, inst, data, length)
+
+/**
+ * @brief Writes an INFO level message to the log.
+ *
+ * @details It's meant to write generic user oriented messages.
+ *
+ * @param inst   Pointer to the log filter associated with the instance.
+ * @param data   Pointer to the data to be logged.
+ * @param length Length of data (in bytes).
+ */
+#define LOG_INST_HEXDUMP_INFO(inst, data, length) \
+	LOG_INTERNAL_HEXDUMP_INSTANCE(LOG_LEVEL_INFO, inst, data, length)
+
+/**
+ * @brief Writes an DEBUG level message to the log.
+ *
+ * @details It's meant to write developer oriented information.
+ *
+ * @param inst   Pointer to the log filter associated with the instance.
+ * @param data   Pointer to the data to be logged.
+ * @param length Length of data (in bytes).
+ */
+#define LOG_INST_HEXDUMP_DBG(inst, data, length) \
+	LOG_INTERNAL_HEXDUMP_INSTANCE(LOG_LEVEL_DBG, inst, data, length)
+
+/**
+ * @brief Writes an formatted string to the log.
+ *
+ * @details Conditionally compiled (@ref CONFIG_LOG_PRINTK). Function provides
+ * printk functionality. It is ineffective compared to standard logging
+ * because string formatting is performed in the call context and not deferred to
```

Comment:
> not deferred to the known context.

What is meant by "known" here? I guess, it should be "logging backend context".

Reply:
rephrased.